src/nyx/scan_fastloader_way.cpp

```diff
@@ -367,6 +452,19 @@ namespace Nyxus
 			if (PyErr_CheckSignals() != 0)
                 		throw pybind11::error_already_set();
 		}
+
+	#ifdef USE_ARROW
+		if (arrow_output) {
+			// close arrow file after use
+			auto status = theEnvironment.arrow_writer->close();
+			
+			if (!status.ok()) {
+				// Handle read error
+				error_message = "Error closing Arrow file: " + status.ToString();
+				return 2;
+			}
+		}
+	#endif
```

Comment:
File scan_fastloader_way.cpp is getting cluttered with #ifdef USE_ARROW code. Which is OK for this PR but I would immediately fix in a subsequent PR to keep the image scan logic free of low-level file format stuff. (Principle of separation of concerns.)